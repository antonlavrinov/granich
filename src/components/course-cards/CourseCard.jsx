import React from 'react'
import {
    ArrowIcon,
    // BlackArrow,
    CourseWrapper,
    CourseContainer,
    CourseButtonContainer,
    CourseTagList,
    CourseMainTag,
    CourseTag,
    CourseTitle,
    CourseImage,
    CourseImageWrapper,
    CourseButton,
    CourseButtonWrapper,
    CourseButtonText,
    CourseInfo,
    CourseStart,
    CourseDuration,
    CourseTeachers,
    CourseTeacher,
    // Calendar
} from './styles';



const CourseCard = ({ courseData, empty, calendar, arrowWhite, arrowBlack }) => {

    return (
        <CourseWrapper>
            <CourseContainer>
                <CourseTagList empty={empty}>
                    <>
                        {courseData.node.courseTags.map((courseTag, idx) => {
                            return (
                                <CourseTag key={idx} active={courseData.node.courseStatus}>{courseTag}</CourseTag>
                            )
                        })}

                    </>


                </CourseTagList>

                <CourseImageWrapper>
                    <CourseImage empty={empty} active={courseData.node.courseStatus} fluid={courseData.node.coursePreviewImage.fluid}>
                    </CourseImage>
                    <CourseMainTag empty={empty} active={courseData.node.courseStatus}>Курс</CourseMainTag>
                    <CourseTeachers>
                        {courseData.node.courseTeachers.length > 1 ? (
                            <>
                                {courseData.node.courseTeachers.map((teacher, idx) => {
                                    return (
                                        <CourseTeacher several={true} key={idx} fluid={teacher.fluid} />
                                    )
                                })}
                            </>

                        ) : (
                                <CourseTeacher fluid={courseData.node.courseTeachers[0].fluid} />
                            )}
                    </CourseTeachers>
                </CourseImageWrapper>
                <CourseTitle active={courseData.node.courseStatus}>{courseData.node.courseTitle}</CourseTitle>
                <CourseInfo active={courseData.node.courseStatus}>
                    {courseData.node.courseStart && (
                        <CourseStart active={courseData.node.courseStatus}>
                            Стартуем <br /> <span>{courseData.node.courseStart}</span>
                        </CourseStart>
                    )}
                    {courseData.node.courseDuration && (
                        <CourseDuration active={courseData.node.courseStatus}>
                            Учимся <br /> <span>{courseData.node.courseDuration}</span>
                        </CourseDuration>
                    )}
                </CourseInfo>

            </CourseContainer>
            <CourseButton empty={empty ? 1 : 0} active={courseData.node.courseStatus}>
                <CourseButtonContainer>
                    <CourseButtonWrapper>
                        {empty ? (
                            <>
                                <CourseButtonText empty>
                                    Узнать <br />об открытии
                                </CourseButtonText>
                                <ArrowIcon fluid={calendar} />
                            </>
                        ) : (
                                <>
                                    {courseData.node.courseStatus ? (
                                        <>

                                            <CourseButtonText>
                                                Участвовать
                                        </CourseButtonText>
                                            <ArrowIcon fluid={arrowWhite} />

                                        </>
                                    ) : (
                                            <>
                                                <CourseButtonText>
                                                    Узнать <br />о наборе
                                        </CourseButtonText>
                                                <ArrowIcon fluid={arrowBlack} />
                                            </>
                                        )}
                                </>
                            )}


                    </CourseButtonWrapper>

                </CourseButtonContainer>
            </CourseButton>
        </CourseWrapper>
    )
}

export default CourseCard
