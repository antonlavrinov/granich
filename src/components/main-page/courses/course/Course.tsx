import React from "react"
import styled from "styled-components"
import BackgroundImage from "gatsby-image"
import CogwheelIcon from "../../../../assets/svgs/cogwheels.svg"
// import CalendarIcon from '../../assets/svgs/empty_calendar.svg';
import Img from "gatsby-image"

export const LinkWrapper = styled.a`
  transition: all 0.2s ease;
  border-radius: 0.6vw;
  // box-shadow: 0 0 0.9vw rgba(0,0,0,0.25);
  :hover {
    cursor: pointer;
    transform: scale(1.03);
  }
  @media only screen and (max-width: 575px) {
    border-radius: 1.5vw;
    // box-shadow: 0 0 1.5vw rgba(0,0,0,0.25);
    :hover {
      transform: none;
    }
  }
`

export const LinkModal = styled.div`
  // text-align: left;
  // background: none;
  // border-radius: 0.6vw;
  // box-shadow: 0 0 0.9vw rgba(0,0,0,0.25);
  :hover {
    cursor: pointer;
  }

  ${props => props.clickable && `
    :hover {
      cursor: default;
    }
    

  `}
  // @media only screen and (max-width: 575px) {
  //     // border-radius: 1.5vw;
  //     // box-shadow: 0 0 1.5vw rgba(0,0,0,0.25);
  // }
`

export const Cogwheel = styled(props => <CogwheelIcon {...props} />)`
  width: 5vw;
  height: 5vw;
  position: absolute;
  top: 50%;
  right: 50%;
  transform: translate(50%, -50%);
`

export const ArrowIcon = styled(props => <Img {...props} />)`
  width: 3.1vw;
  @media only screen and (max-width: 575px) {
    width: 7.5vw;
    margin-right: -0.5vw;
  }
`
// export const BlackArrow = styled(props => <BlackArrowNewPage {...props}/>)`
//     width: 3.1vw;
//     @media only screen and (max-width: 575px) {
//         width: 7.5vw;
//         margin-right: -0.5vw;
//     }
// `

// export const Calendar = styled(props => <CalendarIcon {...props}/>)`
//     width: 3.1vw;
//     @media only screen and (max-width: 575px) {
//         width: 7.5vw;
//         margin-right: -0.5vw;
//     }
// `

export const Wrapper = styled.div`
  background: white;
  border-radius: 0.6vw;
  box-shadow: 0 0 0.9vw rgba(0, 0, 0, 0.25);
  display: flex;
  flex-direction: column;
  height: 100%;

  ${props =>
    props.development &&
    `
        background: #E8E8E8;
        box-shadow: none;
        :hover {
            cursor: default;
            transform: none;
        }
        @media only screen and (max-width: 575px) {
            display: none;
        }
    `}
  @media only screen and (max-width: 575px) {
    border-radius: 1.5vw;
    // box-shadow: 0 0 1.5vw rgba(0,0,0,0.25);
  }
`

export const DevelopingWrapper = styled.div`
  background: #e8e8e8;
  border-radius: 0.6vw;
  display: flex;
  flex-direction: column;
  @media only screen and (max-width: 575px) {
    display: none;
  }
`

export const Container = styled.div`
  padding: 0.9vw 1vw;
  @media only screen and (max-width: 575px) {
    padding: 2.5vw 2.5vw 3.5vw;
  }
`
export const ButtonContainer = styled.div`
  padding: 0.9vw;
  display: flex;
  flex-direction: column;
  height: 100%;
  ${props =>
    props.development &&
    `
        padding: 2vw 0.9vw;
    `}

  @media only screen and (max-width: 575px) {
    // padding: 3vw 5vw 3vw;
    padding: 2.5vw;
  }
`

export const TagList = styled.div`
  margin-bottom: 0.7vw;
  display: flex;
  align-items: flex-start;
  width: 15.4vw;

  @media only screen and (max-width: 575px) {
    width: 38vw;
    margin-bottom: 1.5vw;
    margin-top: 0;
    margin-left: -0.2vw;
  }
  ${props =>
    props.empty &&
    `
        div {
            color: var(--granich-grey);
            border-color: var(--granich-grey);
        }
    `}
`

export const MainTag = styled.div`
  border: 1px solid var(--granich-red);
  color: white;
  border-radius: 100vw;
  background: var(--granich-red);
  font-weight: 600;
  line-height: 1;
  font-size: 0.9vw;
  padding: 0.2vw 0.5vw 0.2vw;
  display: inline-block;
  letter-spacing: -0.05vw;
  position: absolute;
  bottom: 0.7vw;
  left: 0.7vw;
  z-index: 101;
  ${props =>
    !props.active &&
    `
        background: var(--granich-black);
        border-color: var(--granich-black);

    `}
  @media only screen and (max-width: 575px) {
    font-size: 2.5vw;
    padding: 0.2vw 1vw 0.2vw;
    border-width: 0.3vw;
    bottom: 1vw;
    left: 1vw;
  }
  ${props =>
    props.empty &&
    `
        background: var(--granich-black);
        border-color: var(--granich-black);
    `}
`

export const Tag = styled.div`
  border: 1px solid var(--granich-red);
  color: var(--granich-red);
  border-radius: 100vw;
  font-weight: 500;
  line-height: 1;
  font-size: 1vw;
  padding: 0.2vw 0.5vw 0.2vw;
  display: block;
  letter-spacing: -0.03vw;
  margin-right: 0.4vw;
  white-space: nowrap;
  :last-child {
    margin-right: 0;
  }
  ${props =>
    !props.active &&
    `
        // filter: opacity(40%);
        border-color: var(--granich-grey);
        color: var(--granich-grey);

    `}
  ${props =>
    props.development &&
    `
        border: 2px solid #d4d4d4;
        filter: none;
        width: 100%;
        height: 1.45vw;

    `}
    @media only screen and (max-width: 575px) {
    border-width: 1px;
    font-size: 2.6vw;
    padding: 0.7vw 0.8vw;
    font-weight: 400;
    margin-right: 0.6vw;
    letter-spacing: -0.02em;
  }
`

export const Title = styled.h3`
  font-size: 1.75vw;
  margin-top: 0.4vw;
  margin-left: 0.3vw;
  line-height: 1.05;
  font-weight: 600;
  letter-spacing: 0;
  width: 15vw;
  @media only screen and (max-width: 575px) {
    font-size: 3.8vw;
    margin-top: 3vw;
    margin-left: 0.25vw;
    line-height: 1;
    width: 37vw;
  }
`

export const Image = styled(props => (
  <BackgroundImage {...props}></BackgroundImage>
))`
  min-height: 15.5vw;
  width: 100%;
  position: relative;
  @media only screen and (max-width: 575px) {
    min-height: 38vw;
    max-height: 38vw;
    height: 38vw;
  }
  ${props =>
    (props.empty || !props.active) &&
    `
        filter: grayscale(100%);
    `}
`

export const ImageWrapper = styled.div`
  position: relative;
`

export const Button = styled.div`
    background: var(--granich-red);
    border-radius: 0 0 0.6vw 0.6vw;
    color: white;
    margin-top: auto;
    @media only screen and (max-width: 575px) {
        border-radius: 0 0 1.5vw 1.5vw;
    }
    ${props =>
      !props.active &&
      `
        background: black;
    
    `}
    ${props =>
      props.development &&
      `
        background: #dedede;
    `}
    ${props =>
      props.empty &&
      `
        background: black;
    `}

    
`

export const ButtonWrapper = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`
export const ButtonText = styled.div`
  color: white;
  font-size: 1.4vw;
  font-weight: 500;
  line-height: 0.95;
  margin-left: 0.3vw;
  margin-top: -0.2vw;
  ${props =>
    props.development &&
    `
        color: #C0C0C0;
    `}
  ${props =>
    props.empty &&
    `
        line-height: 1;
    `}
    @media only screen and (max-width: 575px) {
    font-size: 3.5vw;
    font-size: 4vw;
    margin-top: -1vw;
  }
`

export const Info = styled.div`
  display: flex;
  margin-left: 0.3vw;
  font-size: 1vw;
  line-height: 1.1;
  margin-bottom: 0.45vw;
  margin-top: 0.9vw;
  ${props =>
    !props.active &&
    `
        color: var(--granich-black);
    `}
  @media only screen and (max-width: 575px) {
    font-size: 2.7vw;
    font-size: 3vw;
    margin-top: 2.5vw;
    margin-bottom: 0vw;
    margin-left: 0vw;
  }
`
export const Descr = styled.div`
  font-size: 1vw;
  margin-left: 0.3vw;
  color: var(--granich-grey);
  letter-spacing: -0.01vw;
  font-weight: 500;
  line-height: 1;
  margin-top: 0.5vw;
  width: 15vw;
  ${props =>
    !props.active &&
    `
        color: var(--granich-light-grey);
    `}
  @media only screen and (max-width: 575px) {
    margin-left: 0.1vw;
    font-size: 2.7vw;
    font-size: 3vw;
    margin-top: 1vw;
    width: 37vw;
  }
`

export const Start = styled.div`
  font-size: inherit;
  margin-right: 1.1vw;
  font-weight: 500;
  color: var(--granich-light-grey);
  span {
    font-weight: inherit;
    font-size: inherit;
    color: black;
  }
  @media only screen and (max-width: 575px) {
    margin-right: 3vw;
  }
`

export const Duration = styled.div`
  font-size: inherit;
  font-weight: 500;
  color: var(--granich-light-grey);
  max-width: 9.2vw;
  span {
    font-weight: inherit;
    font-size: inherit;
    color: black;
    max-width: 20.2vw;
  }
`

export const Teachers = styled.div`
  position: absolute;
  display: flex;
  justify-content: flex-end;
  bottom: 0;
  left: 0;
  width: 100%;
  padding: 0.9vw 0.6vw 0.6vw;
  background: linear-gradient(
    360deg,
    rgba(0, 0, 0, 0.2) 0%,
    rgba(45, 45, 45, 0) 80%
  );
  z-index: 100;
  @media only screen and (max-width: 575px) {
    padding: 1vw;
  }
`
export const Teacher = styled(props => (
  <BackgroundImage {...props}></BackgroundImage>
))`
  border-radius: 100vw;
  margin-top: -0.2vw;
  margin-right: 0;
  height: 2.35vw;
  width: 2.35vw;
  border: 1px solid var(--granich-black);
  box-shadow: 0.15vw 0.12vw 0.2vw rgba(0, 0, 0, 0.25);
  background: var(--granich-grey);
  @media only screen and (max-width: 575px) {
    height: 5.8vw;
    width: 5.8vw;
  }
  ${props =>
    props.several &&
    `
        margin-right: -0.2vw;
        :last-child {
            margin-right: 0;
            z-index: 1;
        }
        @media only screen and (max-width: 575px) {
            margin-right: -1vw;
            :last-child {
                margin-right: 0;
            }
        }
    `}
`

export const DevelopingImage = styled.div`
  min-height: 15.5vw;
  border: 2px solid #d4d4d4;
  margin-bottom: 0.7vw;
  position: relative;
`
export const DevelopingTitle = styled.div`
  div {
    height: 1.3vw;
    width: 70%;
    background: #dedede;
    margin-bottom: 0.4vw;
    :last-child {
      margin-bottom: 0;
      width: 45%;
    }
  }
  margin-bottom: 0.4vw;
`
export const DevelopingDescr = styled.div`
  div {
    height: 0.7vw;
    width: 90%;
    background: #dedede;
    margin-bottom: 0.3vw;
    :nth-child(2) {
      width: 55%;
    }
    :last-child {
      margin-bottom: 0;
      width: 70%;
    }
  }
`
