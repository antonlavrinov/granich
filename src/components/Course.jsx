import React from 'react'
import styled from 'styled-components';
import CourseCard from './course-cards/CourseCard';
import MasterClassCard from './course-cards/MasterClassCard';
import DevelopmentCard from './course-cards/DevelopmentCard';
import { graphql, useStaticQuery } from 'gatsby'
import { trackCustomEvent } from 'gatsby-plugin-google-analytics'



const CourseLinkWrapper = styled.a`
    transition: all 0.2s ease;
    :hover {
        cursor: pointer;
        transform: scale(1.03);
    }
    @media only screen and (max-width: 575px) {
        :hover {
            transform: none;
        }
    }
`

const CourseLinkModal = styled.div`
    text-align: left;
    background: none;
    :hover {
        cursor: pointer;
    }

`




const Course = ({ courseData, openModal }) => {
    const data = useStaticQuery(graphql`
        query courseIcons {
            courseCalendar: file(relativePath: { eq: "empty_calendar-01.png" }) {
                childImageSharp {
                    fluid(maxWidth: 120) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }
            courseArrowBlack: file(relativePath: { eq: "arrow-black-newpage-01.png" }) {
                childImageSharp {
                    fluid(maxWidth: 120) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }
            courseArrowWhite: file(relativePath: { eq: "arrow-white-newpage-01.png" }) {
                childImageSharp {
                    fluid(maxWidth: 120) {
                        ...GatsbyImageSharpFluid
                    }
                }
            }


        }
    `)

    const calendar = data.courseCalendar.childImageSharp.fluid;
    const arrowWhite = data.courseArrowWhite.childImageSharp.fluid;
    const arrowBlack = data.courseArrowBlack.childImageSharp.fluid;


    if (courseData.node.courseTitle === 'Графсистемы Брутализм и Антидизайн') {
        return null
    }

    console.log('COURSE TITLE', courseData.node.courseTitle)

    return (
        <>
            {courseData.node.courseTypeDevelopment ? (
                <DevelopmentCard />
            ) : (
                    <>
                        {courseData.node.courseTypeEmpty ? (
                            <CourseLinkModal type="button" onClick={() => { openModal(); trackCustomEvent({ category: `Главная: карточка "${courseData.node.courseTitle}"`, action: 'click' }) }} >
                                {courseData.node.courseType === 'Курс' ? (
                                    <CourseCard arrowBlack={arrowBlack} arrowWhite={arrowWhite} calendar={calendar} empty courseData={courseData} />
                                ) : (
                                        <MasterClassCard arrowBlack={arrowBlack} arrowWhite={arrowWhite} calendar={calendar} empty courseData={courseData} />
                                    )}
                            </CourseLinkModal>
                        ) : (
                                <CourseLinkWrapper onClick={() => trackCustomEvent({ category: `Главная: карточка "${courseData.node.courseTitle}"`, action: 'click' })} href={`/${courseData.node.courseSlug}`}>
                                    <>
                                        {courseData.node.courseType === 'Курс' ? (
                                            <CourseCard arrowBlack={arrowBlack} arrowWhite={arrowWhite} calendar={calendar} courseData={courseData} />
                                        ) : (
                                                <MasterClassCard arrowBlack={arrowBlack} arrowWhite={arrowWhite} calendar={calendar} courseData={courseData} />
                                            )}
                                    </>
                                </CourseLinkWrapper>
                            )}

                    </>
                )}

        </>

    )
}

export default Course;