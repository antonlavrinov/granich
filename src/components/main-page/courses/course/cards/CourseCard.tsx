import React from "react"
import * as SC from "../Course"

const CourseCard = ({
  courseData,
  empty,
  calendar,
  arrowWhite,
  arrowBlack,
}) => {
  const mainTagMentorType =
    courseData.node.courseTitle === "Осознанный Фриланс"
      ? "-наставничество"
      : ""

  return (
    <SC.Wrapper>
      <SC.Container>
        <SC.TagList empty={empty}>
          <>
            {courseData.node.courseTags.map((courseTag, idx) => {
              return (
                <SC.Tag key={idx} active={courseData.node.courseStatus}>
                  {courseTag}
                </SC.Tag>
              )
            })}
          </>
        </SC.TagList>

        <SC.ImageWrapper>
          <SC.Image
            empty={empty}
            active={courseData.node.courseStatus}
            fluid={courseData.node.coursePreviewImage.fluid}
          ></SC.Image>
          <SC.MainTag empty={empty} active={courseData.node.courseStatus}>
            Курс{mainTagMentorType}
          </SC.MainTag>
          <SC.Teachers>
            {courseData.node.courseTeachers.length > 1 ? (
              <>
                {courseData.node.courseTeachers.map((teacher, idx) => {
                  return (
                    <SC.Teacher
                      several={true}
                      key={idx}
                      fluid={teacher.fluid}
                    />
                  )
                })}
              </>
            ) : (
              <SC.Teacher fluid={courseData.node.courseTeachers[0].fluid} />
            )}
          </SC.Teachers>
        </SC.ImageWrapper>
        <SC.Title active={courseData.node.courseStatus}>
          {courseData.node.courseTitle}
        </SC.Title>
        <SC.Info active={courseData.node.courseStatus}>
          {courseData.node.courseStart && (
            <SC.Start active={courseData.node.courseStatus}>
              Стартуем <br /> <span>{courseData.node.courseStart}</span>
            </SC.Start>
          )}
          {courseData.node.courseDuration && (
            <SC.Duration active={courseData.node.courseStatus}>
              Учимся <br /> <span>{courseData.node.courseDuration}</span>
            </SC.Duration>
          )}
        </SC.Info>
      </SC.Container>
      <SC.Button empty={empty ? 1 : 0} active={courseData.node.courseStatus}>
        <SC.ButtonContainer>
          <SC.ButtonWrapper>
            {empty ? (
              <>
                <SC.ButtonText empty>
                  Узнать <br />
                  об открытии
                </SC.ButtonText>
                <SC.ArrowIcon fluid={calendar} />
              </>
            ) : (
              <>
                {courseData.node.courseStatus ? (
                  <>
                    <SC.ButtonText>Участвовать</SC.ButtonText>
                    <SC.ArrowIcon fluid={arrowWhite} />
                  </>
                ) : (
                  <>
                    <SC.ButtonText>
                      Узнать <br />о наборе
                    </SC.ButtonText>
                    <SC.ArrowIcon fluid={arrowBlack} />
                  </>
                )}
              </>
            )}
          </SC.ButtonWrapper>
        </SC.ButtonContainer>
      </SC.Button>
    </SC.Wrapper>
  )
}

export default CourseCard
