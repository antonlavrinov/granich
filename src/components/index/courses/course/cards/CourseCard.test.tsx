import React from "react"
import { shallow } from "enzyme"
import CourseCard from "./CourseCard"
import { ButtonText } from "../Course"
import { mockCourseData, mockIcons } from "../Course.test"

describe("<CourseCard> component", () => {
  describe("Active ", () => {
    const wrapper = shallow(
      <CourseCard courseData={mockCourseData} icons={mockIcons} />
    )
    it("should render correctly", () => {
      expect(wrapper).toMatchSnapshot()
    })
    it("should have button text 'Участвовать'", () => {
      expect(wrapper.find(ButtonText).text()).toBe("Участвовать")
    })
  })
  describe("Not Active ", () => {
    const wrapper = shallow(
      <CourseCard
        courseData={{ ...mockCourseData, courseStatus: false }}
        icons={mockIcons}
      />
    )
    it("should render correctly", () => {
      expect(wrapper).toMatchSnapshot()
    })
    it("should have button text 'Узнать о наборе'", () => {
      expect(wrapper.find(ButtonText).text()).toBe("Узнать о наборе")
    })
  })
})
