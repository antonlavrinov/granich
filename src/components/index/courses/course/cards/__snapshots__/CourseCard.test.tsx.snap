// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Course> component Course card Empty should render correctly 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
  >
    <CourseCard
      courseData={
        Object {
          "courseDescr": "some descr",
          "courseDuration": "some date",
          "coursePreviewImage": Object {
            "fluid": "imageUrl",
          },
          "courseSlug": "slug",
          "courseStart": "course start",
          "courseStatus": true,
          "courseTags": Array [
            "1",
            "2",
          ],
          "courseTeachers": Array [
            Object {
              "fluid": "imageUrl",
              "id": 3,
            },
            Object {
              "fluid": "imageUrl",
              "id": 4,
            },
          ],
          "courseTitle": "title",
          "courseType": "Курс",
          "courseTypeDevelopment": false,
          "courseTypeEmpty": true,
          "id": 0,
        }
      }
      icons={
        Object {
          "arrowBlack": "arrowBlackIcon",
          "arrowWhite": "arrowWhiteIcon",
          "calendar": "calendarIcon",
        }
      }
    />
  </styled.div>
</Fragment>
`;

exports[`<Course> component Course card should render correctly 1`] = `
<Fragment>
  <styled.a
    href="/slug"
    onClick={[Function]}
  >
    <CourseCard
      courseData={
        Object {
          "courseDescr": "some descr",
          "courseDuration": "some date",
          "coursePreviewImage": Object {
            "fluid": "imageUrl",
          },
          "courseSlug": "slug",
          "courseStart": "course start",
          "courseStatus": true,
          "courseTags": Array [
            "1",
            "2",
          ],
          "courseTeachers": Array [
            Object {
              "fluid": "imageUrl",
              "id": 3,
            },
            Object {
              "fluid": "imageUrl",
              "id": 4,
            },
          ],
          "courseTitle": "title",
          "courseType": "Курс",
          "courseTypeDevelopment": false,
          "courseTypeEmpty": false,
          "id": 0,
        }
      }
      icons={
        Object {
          "arrowBlack": "arrowBlackIcon",
          "arrowWhite": "arrowWhiteIcon",
          "calendar": "calendarIcon",
        }
      }
    />
  </styled.a>
</Fragment>
`;

exports[`<Course> component Development card should render correctly 1`] = `
<Fragment>
  <DevelopmentCard />
</Fragment>
`;

exports[`<Course> component MasterClass card Empty should render correctly 1`] = `
<Fragment>
  <styled.div
    onClick={[Function]}
  >
    <MasterCard
      courseData={
        Object {
          "courseDescr": "some descr",
          "courseDuration": "some date",
          "coursePreviewImage": Object {
            "fluid": "imageUrl",
          },
          "courseSlug": "slug",
          "courseStart": "course start",
          "courseStatus": true,
          "courseTags": Array [
            "1",
            "2",
          ],
          "courseTeachers": Array [
            Object {
              "fluid": "imageUrl",
              "id": 3,
            },
            Object {
              "fluid": "imageUrl",
              "id": 4,
            },
          ],
          "courseTitle": "title",
          "courseType": "Мастер-класс",
          "courseTypeDevelopment": false,
          "courseTypeEmpty": true,
          "id": 0,
        }
      }
      icons={
        Object {
          "arrowBlack": "arrowBlackIcon",
          "arrowWhite": "arrowWhiteIcon",
          "calendar": "calendarIcon",
        }
      }
    />
  </styled.div>
</Fragment>
`;

exports[`<Course> component MasterClass card should render correctly 1`] = `
<Fragment>
  <styled.a
    href="/slug"
    onClick={[Function]}
  >
    <MasterCard
      courseData={
        Object {
          "courseDescr": "some descr",
          "courseDuration": "some date",
          "coursePreviewImage": Object {
            "fluid": "imageUrl",
          },
          "courseSlug": "slug",
          "courseStart": "course start",
          "courseStatus": true,
          "courseTags": Array [
            "1",
            "2",
          ],
          "courseTeachers": Array [
            Object {
              "fluid": "imageUrl",
              "id": 3,
            },
            Object {
              "fluid": "imageUrl",
              "id": 4,
            },
          ],
          "courseTitle": "title",
          "courseType": "Мастер-класс",
          "courseTypeDevelopment": false,
          "courseTypeEmpty": false,
          "id": 0,
        }
      }
      icons={
        Object {
          "arrowBlack": "arrowBlackIcon",
          "arrowWhite": "arrowWhiteIcon",
          "calendar": "calendarIcon",
        }
      }
    />
  </styled.a>
</Fragment>
`;

exports[`<CourseCard> component Active  should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      empty={false}
    >
      <styled.div
        active={true}
        key="0"
      >
        1
      </styled.div>
      <styled.div
        active={true}
        key="1"
      >
        2
      </styled.div>
    </styled.div>
    <styled.div>
      <Styled(Component)
        active={true}
        empty={false}
        fluid="imageUrl"
      />
      <styled.div
        active={true}
        empty={false}
      >
        Курс
      </styled.div>
      <styled.div>
        <Styled(Component)
          fluid="imageUrl"
          key="0"
          several={true}
        />
        <Styled(Component)
          fluid="imageUrl"
          key="1"
          several={true}
        />
      </styled.div>
    </styled.div>
    <styled.h3
      active={true}
    >
      title
    </styled.h3>
    <styled.div
      active={true}
    >
      <styled.div
        active={true}
      >
        Стартуем 
        <br />
         
        <span>
          course start
        </span>
      </styled.div>
      <styled.div
        active={true}
      >
        Учимся 
        <br />
         
        <span>
          some date
        </span>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    active={true}
    empty={0}
  >
    <styled.div>
      <styled.div>
        <styled.div>
          Участвовать
        </styled.div>
        <Styled(Component)
          fluid="arrowWhiteIcon"
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;

exports[`<CourseCard> component Not Active  should render correctly 1`] = `
<styled.div>
  <styled.div>
    <styled.div
      empty={false}
    >
      <styled.div
        active={false}
        key="0"
      >
        1
      </styled.div>
      <styled.div
        active={false}
        key="1"
      >
        2
      </styled.div>
    </styled.div>
    <styled.div>
      <Styled(Component)
        active={false}
        empty={false}
        fluid="imageUrl"
      />
      <styled.div
        active={false}
        empty={false}
      >
        Курс
      </styled.div>
      <styled.div>
        <Styled(Component)
          fluid="imageUrl"
          key="0"
          several={true}
        />
        <Styled(Component)
          fluid="imageUrl"
          key="1"
          several={true}
        />
      </styled.div>
    </styled.div>
    <styled.h3
      active={false}
    >
      title
    </styled.h3>
    <styled.div
      active={false}
    >
      <styled.div
        active={false}
      >
        Стартуем 
        <br />
         
        <span>
          course start
        </span>
      </styled.div>
      <styled.div
        active={false}
      >
        Учимся 
        <br />
         
        <span>
          some date
        </span>
      </styled.div>
    </styled.div>
  </styled.div>
  <styled.div
    active={false}
    empty={0}
  >
    <styled.div>
      <styled.div>
        <styled.div>
          Узнать 
          <br />
          о наборе
        </styled.div>
        <Styled(Component)
          fluid="arrowBlackIcon"
        />
      </styled.div>
    </styled.div>
  </styled.div>
</styled.div>
`;
